crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -54,38 +54,46 @@ pub struct Line2d {
 }
 impl Primitive2d for Line2d {}
 
-/// A section of a line along a direction in 2D space.
+/// A segment of a line along a direction in 2D space.
+#[doc(alias = "LineSegment2d")]
 #[derive(Clone, Debug)]
-pub struct LineSegment2d {
+pub struct Segment2d {
     /// The direction of the line
     pub direction: Direction2d,
     /// Half the length of the line segment, the segment extends by this amount in both
-    /// the in both the positive and negative direction
+    /// the positive and negative direction
     pub half_length: f32,
 }
-impl Primitive2d for LineSegment2d {}
+impl Primitive2d for Segment2d {}
 
-impl LineSegment2d {
-    /// Get a line segment and translation from a start and end position of a line segment
-    pub fn from_start_end(start: Vec2, end: Vec2) -> (Self, Vec2) {
-        let diff = end - start;
+impl Segment2d {
+    /// Create a line segment from a direction and full length of the line
+    pub fn new(direction: Direction2d, length: f32) -> Self {
```

Comment:
nit: ask what you need, and don't assume what the user have; that will avoid them having to pass `half_length * 2` only to be divided again.

```suggestion
    pub fn new(direction: Direction2d, half_length: f32) -> Self {
```

Reply:
Main reason I made it like that is because it's not all that common start with a half length, and even if you did the struct fields are public and don't have the footgun of not realizing its a half length (because you need to type out the field name). But maybe having it be called `new` and then taking a different format is kind of confusing either way

Reply:
Agreed: because the fields are public, the constructor methods should be intuitive.

Reply:
Fair enough